compiler-core/src/language_server/engine.rs

```diff
@@ -452,43 +473,69 @@ where
                     continue;
                 }
 
-                let module = import.used_name();
-                if module.is_some() {
-                    completions.push(type_completion(module.as_ref(), name, type_));
+                if let Some(module) = import.used_name() {
+                    // If the user has already started a module select then don't show irrelevant modules.
+                    if let Some(input_mod_name) = &module_select {
+                        if !module.contains(input_mod_name.as_str()) {
```

Comment:
Contains rather than equals?

Reply:
I used contains here to support typing partial pieces of the module name which I could see people doing a lot (e.x. if I had an `unordered_set` module a user could type `set.` and still get relevant results). lmk what you think I can just as easily switch it to equals

Reply:
Let's try equality and we can relax it later if it feels bad